plugins/blockvolume/hostvol/utils.go

```diff
@@ -15,34 +15,70 @@ import (
 	"github.com/gluster/glusterd2/glusterd2/volume"
 	"github.com/gluster/glusterd2/pkg/api"
 	"github.com/gluster/glusterd2/pkg/size"
-
 	log "github.com/sirupsen/logrus"
 )
 
+func filterSize(volinfo *volume.Volinfo, size uint64) bool {
+	availableSize, found := volinfo.Metadata[volume.BlockHostingAvailableSize]
+	if !found {
+		return false
+	}
+	if availableSizeInBytes, err := strconv.ParseUint(availableSize, 10, 64); err == nil && availableSizeInBytes > size {
+		return true
+	}
+
+	return false
+}
+
+func applyArbiter(volinfo *volume.Volinfo, arbPath string) bool {
+	if arbPath == "" {
+		return true
+	}
+	_, exists := volinfo.Options["cluster/replicate.thin-arbiter"]
+	if exists {
+		return true
+	}
+	return false
+}
+
+func applyShard(volinfo *volume.Volinfo, size uint64) bool {
+	if size == 0 {
+		return true
+	}
+	_, exists := volinfo.Options["features/shard"]
+	if exists {
+		//Checking for shard size may not be required
+		return true
+	}
+	return false
+}
+
 // BlockSizeFilter returns a volume Filter, which will filter out volumes
 // haing block-hosting-available-size greater than give size.
-func BlockSizeFilter(size uint64) volume.Filter {
+func HostOptnsFilter(size uint64, h *HostingVolumeOptions) volume.Filter {
```

Comment:
write applyArbiter and applyShard separately as a filter function and this function can be omitted.

Reply:
you can add  FilterArbiter and FilterShard func in this file `github.com/gluster/glusterd2/glusterd2/volume/filters.go`

Reply:
Done